### ISSUE TITLE
Working on iOS-- not on Android

### ISSUE THREAD

**Issue by GolferGeek** [sitefinitysteve/nativescript-auth0#51]
I'm working in Angular and got things to work in iOS.  When running in Android I get this:  "Aw snap! Something went wrong while displaying this webpage."

I played around some with data element in the manifest.xml:

				<data android:host="lubetech.auth0.com" android:pathPrefix="/android/lubetech.nativescript.shop/callback" android:scheme="https" />

I checked to make sure that this was in the callback.

Just running in an emulator... Nexus S-- Api 28 (Pie)

CLI:   5.1.1
TNS Modules: 5.1.0
TNS Android: 5.1.0
TNS iOS: 5.1.1

Plugins:
"dependencies": {
        "@angular/animations": "~7.1.0",
        "@angular/common": "~7.1.0",
        "@angular/compiler": "~7.1.0",
        "@angular/core": "~7.1.0",
        "@angular/forms": "~7.1.0",
        "@angular/http": "~7.1.0",
        "@angular/platform-browser": "~7.1.0",
        "@angular/platform-browser-dynamic": "~7.1.0",
        "@angular/router": "~7.1.0",
        "nativescript-angular": "~7.1.0",
        "nativescript-theme-core": "~1.0.4",
        "nativescript-ui-sidedrawer": "~5.1.0",
        "nativescript-auth0": "^2.1.0",
        "reflect-metadata": "~0.1.12",
        "rxjs": "~6.3.0",
        "tns-core-modules": "~5.1.0",
        "zone.js": "~0.8.26"
    },
    "devDependencies": {
        "@angular/compiler-cli": "~7.1.0",
        "@nativescript/schematics": "~0.5.0",
        "@ngtools/webpack": "~7.1.0",
        "codelyzer": "~4.5.0",
        "nativescript-dev-sass": "~1.6.0",
        "nativescript-dev-typescript": "~0.7.0",
        "nativescript-dev-webpack": "~0.19.0",
        "tslint": "~5.11.0",
        "npx": "10.2.0"
    }

To re-create:  Added a login component with a button and a tap event.
(login.component.html)
<StackLayout class="p-20">
    <Button text="Login" (tap)="onTap()"></Button>
    <Label text="{{ authenticationService.message }}" class="t-20 text-center c-black" textWrap="true"></Label>
</StackLayout>

(login.component.ts)
 onTap() {
        this.authenticationService.handleAuth0();
    }

(authentication.service.ts)
import { Injectable } from '@angular/core';

import { Auth0, Credentials } from 'nativescript-auth0';

@Injectable({
    providedIn: 'root'
})
export class AuthenticationService {
    message = '';
    requestedScopes =
        'openid profile email ship-to administrate:all price:part place:order impersonate:all-ship-tos view:cost-price';

    accessToken = '';
    private auth0: Auth0;

    constructor() {
        this.auth0 = new Auth0('OtqkvtGzMJjEQfThhWdg65QUYU8QbK8x', 'lubetech.auth0.com');
    }

    handleAuth0() {
        this.auth0
            .webAuthentication({
                scope: this.requestedScopes
            })
            .then((result) => {
                this.getUserFromToken(result);
                console.log(result);
            })
            .catch((e: Error) => console.log(e, e.stack));
    }

    getUserFromToken(auth0result: Credentials) {
        this.message = JSON.stringify(auth0result);
        this.accessToken = auth0result.accessToken;
        this.auth0.getUserInfo(this.accessToken).then((userInfo) => {
            this.message += userInfo;
        });
    }
}

Sending the zip along with this...
[lubetech-shop-ns-ng.zip](https://github.com/sitefinitysteve/nativescript-auth0/files/2857343/lubetech-shop-ns-ng.zip)

**Comment by charsleysa** [comment]
Hi @GolferGeek 

This looks like there might be an issue with the callbacks.

Does this happen all the time or is it intermittent?

**Comment by Nevaan** [comment]
Hi,

I faced some problems with authentication on android too, in my case I was not properly redirected back to application after login (application was trying to open the callback address in browser instead of application) and what worked for me was changing the **scheme** property to other than https (of course in manifest, handleAuth0() method and auth0 settings itself). 

```
handleAuth0() {
    this.auth0
        .webAuthentication({
            scope: this.requestedScopes,
            scheme: "customapp"
        })
        .then((result) => {
            this.getUserFromToken(result);
            console.log(result);
        })
        .catch((e: Error) => console.log(e, e.stack));
}
```

Sorry that I cant provide more precise explanation, but I am not very experienced with mobile development. I suspect that this is related to some default behavior of android OS or its browser, and it is not correctly (for us) recognizing the callback address as application activity, but as regular web page. 

Whats more, I spent a lot of time trying to fix this, when I noticed that the manifest file not always get replaced in result (java) project - it was keeping the old xml, so I had to `tns platform clean android` and it got replaced - but this looks more like nativescript problem itself
